tvb/simulator/models/wong_wang.py

```diff
@@ -39,13 +39,15 @@ def _numba_dfun(S, c, a, b, d, g, ts, w, j, io, dx):
     "Gufunc for reduced Wong-Wang model equations."
 
     if S[0] < 0.0:
-        dx[0] = 0.0 - S[0]
+        S_constraint = 0.0
     elif S[0] > 1.0:
-        dx[0] = 1.0 - S[0]
+        S_constraint = 1.0
     else:
-        x = w[0]*j[0]*S[0] + io[0] + j[0]*c[0]
-        h = (a[0]*x - b[0]) / (1 - numpy.exp(-d[0]*(a[0]*x - b[0])))
-        dx[0] = - (S[0] / ts[0]) + (1.0 - S[0]) * h * g[0]
+        S_constraint = S[0]
```

Comment:
this could again be a SIMD-friendly
```python
S_c = S[0]*(S[0] > 0.0)*(S[0] < 1.0) + (S[0] < 0.0) + (S[0] > 1.0)
```
(but this may not be a big deal performance wise considering the rest of the simulator

Reply:
but then again, I don't understand why you didn't just remove this code here: if you've already constrained S after the integration step, it should always lie on the interval [0, 1].

Reply:
You are write. I removed them.